### ISSUE TITLE
Support clearing radio fields

### ISSUE THREAD

**Issue by coreyward** [sanity-io/sanity#4091]
**Describe the bug**

There is no way to clear radio fields. This causes issues when trying to clear out a nested object to make it empty, and thus have it removed. This is a v3 duplicate of #3109. This is adjacent to #2630, which was mostly resolved in #2718.

**To Reproduce**

1. Select a value for a radio field nested in an optional object field
2. There is no way to clear the radio

**Expected behavior**

Perhaps a “clear” or “x” button to unset the value?

**Comment by KittyGiraudel** [comment]
This is an interesting one because “clearing” a radio group is not something that exists from a HTML perspective. Once defined, it cannot be undone.

**Comment by coreyward** [comment]
Yeah, to my knowledge there is no native UI supplied by browsers for this (as with many things), but setting the `checked` property to a falsey value with JS has the intended effect for vanilla input fields.

**Comment by ifeltsweet** [comment]
@coreyward @KittyGiraudel Sorry to hijack this, but I think if objects could be easily unset with a click of a button that would be great too. Think of it as removing an item from an array. Related to https://github.com/sanity-io/sanity/issues/2630 which I believe has a regression in v3.

**Comment by pedrobonamin** [comment]
Hello all.
We have taken this feedback internally and working on a proposal to solve it, we really appreciate the comments and suggestions!
Will update this issue as soon as the changes are merged.

Thank you!
